vertx-web/src/main/java/io/vertx/ext/web/handler/impl/TemplateHandlerImpl.java

```diff
@@ -30,17 +30,22 @@
   private final TemplateEngine engine;
   private final String templateDirectory;
   private final String contentType;
+  private String indexTemplate;
 
   public TemplateHandlerImpl(TemplateEngine engine, String templateDirectory, String contentType) {
     this.engine = engine;
     this.templateDirectory = templateDirectory;
     this.contentType = contentType;
+    this.indexTemplate = DEFAULT_INDEX_TEMPLATE;
   }
 
   @Override
   public void handle(RoutingContext context) {
 
     String file = templateDirectory + Utils.pathOffset(context.normalisedPath(), context);
+    if (file.endsWith("/")) {
```

Comment:
we might need to have a null check for `indexTemplate` for users who do not want this feature active.